.github/actions/ocm-upgrade/ocm_upgrade.py

```diff
@@ -159,29 +167,24 @@ def retrieve_release_notes(
     version_filter: collections.abc.Callable[[str], bool],
     oci_client: oci.client.Client,
     component_descriptor_lookup: ocm.ComponentDescriptorLookup,
-) -> str | None:
+) -> list[rnm.ReleaseNotesDoc]:
     logger.info(f'fetching release-notes for {upgrade_vector=}')
 
-    release_notes = '\n'.join((
-        rno.release_notes_markdown_with_heading(cid, rn)
-        for cid, rn in rno.release_notes_range_recursive(
-            version_vector=upgrade_vector,
-            component_descriptor_lookup=component_descriptor_lookup,
-            version_lookup=version_lookup,
-            version_filter=version_filter,
-            oci_client=oci_client,
-        )
+    release_notes_docs = list(rno.release_notes_for_vector(
+        upgrade_vector=upgrade_vector,
+        component_descriptor_lookup=component_descriptor_lookup,
+        version_lookup=version_lookup,
+        oci_client=oci_client,
+        version_filter=version_filter,
     ))
 
-    if release_notes:
-        logger.info(f'fetched {len(release_notes)=} characters of release-notes')
-    else:
-        logger.warning('did not find any release-notes')
+    logger.info(f'fetched {len(release_notes_docs)} release-notes documents')
+
+    grouped_release_notes_docs = rno.group_release_notes_docs(release_notes_docs)
```

Comment:
would it make sense to leave the grouping to the caller (separation of concerns..)?